Scenes/Towers/Tower.gd

```diff
@@ -173,6 +173,11 @@ func _ready():
 		_kill_count = _temp_preceding_tower._kill_count
 		_best_hit = _temp_preceding_tower._best_hit
 		_damage_dealt_total = _temp_preceding_tower._damage_dealt_total
+		
+#		Transition all buff groups from preceding tower
+		for group in _temp_preceding_tower.get_groups():
```

Comment:
This transitions not just buff groups but all "groups", which may have unintended consequences in the future. Can you add a separate storage for buff group strings to tower?
```
var _buff_groups: Array = []
func get_buff_groups() -> Array:
func add_buff_group(buff_group: String):
etc
```